packages/checkout/sdk/src/config/remoteTokensFetcher.ts

```diff
@@ -0,0 +1,177 @@
+import { Environment } from '@imtbl/config';
+import { AxiosResponse } from 'axios';
+import {
+  ChainId, ChainSlug, ChainTokensConfig, ImxAddressConfig, TokenInfo,
+} from '../types';
+import { ENV_DEVELOPMENT, IMMUTABLE_API_BASE_URL } from '../env';
+import { HttpClient } from '../api/http';
+import { CheckoutError, CheckoutErrorType } from '../errors';
+import { RemoteConfigFetcher } from './remoteConfigFetcher';
+
+const chainSlugMap = new Map<ChainId, ChainSlug>([
+  [ChainId.IMTBL_ZKEVM_MAINNET, ChainSlug.IMTBL_ZKEVM_MAINNET],
+  [ChainId.IMTBL_ZKEVM_TESTNET, ChainSlug.IMTBL_ZKEVM_TESTNET],
+  [ChainId.IMTBL_ZKEVM_DEVNET, ChainSlug.IMTBL_ZKEVM_DEVNET],
+]);
+
+type TokensEndpointResult = {
+  chain: {
+    id: string;
+    name: string;
+  };
+  contract_address: string;
+  decimals: number;
+  image_url: string | null;
+  is_canonical: boolean;
+  name: string;
+  symbol: string;
+  root_chain_id: string | null;
+  root_contract_address: string | null;
+};
+
+type TokensEndpointResponse = {
+  result: TokensEndpointResult[];
+};
+
+export type RemoteConfigParams = {
+  isDevelopment: boolean;
+  isProduction: boolean;
+};
+
+export class RemoteTokensFetcher {
```

Comment:
It'll sound picky:

What should be a better name for this service? Maybe we can find out how we refer to this api across the company and use a name that matches?

Reply:
feel free to disregard